File: projects/pattern/dsa.cpp
#include <iostream>
using namespace std;
int main()
{
    int m;
    cout << "Enter Any Number : ";
    cin >> m;
    int n = (m + 1) / 2;
    for (int i = 1; i <= n; i++)
    {
        for (int j = 1; j <= n - 1; j++)
        {
            cout << "* ";
        }
        // spaces
        for (int j = 1; j <= 2 * i; j++)
        {
            cout << " ";
        }
        // right star
        for (int j = 1; j <= n - 1; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
    for (int i = n - 1; i >= 1; i--)
    {
        for (int j = 1; j <= n - 1; j++)
        {
            cout << "* ";
        }
        // spaces
        for (int j = 1; j <= 2 * i; j++)
        {
            cout << " ";
        }
        // right star
        for (int j = 1; j <= n - 1; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    return 0;
}

File: projects/pattern/dsa.java
import java.util.Scanner;

public class dsa {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter any Number : ");
        int n = sc.nextInt();

        for (int i = 1; i <= n; i++) {

            // Left triangle
            for (int j = 1; j <= i; j++) {
                System.out.print("* ");
            }

            // Middle spaces
            for (int j = 1; j <= 2 * (n - i); j++) {
                System.out.print("  ");
            }

            // Right triangle
            for (int j = 1; j <= i; j++) {
                System.out.print("* ");
            }

            System.out.println();
        }
    }
}
 
